flake.nix

```diff
@@ -254,6 +254,39 @@
                   pkgs.darwin.apple_sdk.frameworks.CoreFoundation
                 ];
               };
+
+            # Shell script to show what versions of the Holochain tools are installed.
+            # It can be included as a package into the dev shell and is then available by its name - `hn-introspect`.
+            hn-introspect = pkgs.writeShellScriptBin "hn-introspect" ''
+              #!/usr/bin/env bash
+
+              if ! type "hc-scaffold" > /dev/null; then
+                echo "hc-scaffold   : not installed"
+              else
+                echo "hc-scaffold   : $(hc-scaffold --version)"
+              fi
+
+              if ! type "hc-launch" > /dev/null; then
+                echo "hc-launch     : not installed"
+              else
+                echo "hc-launch     : $(hc-launch --version)"
+              fi
+
+              if ! type "lair-keystore" > /dev/null; then
+                echo "Lair keystore : not installed"
+              else
+                echo "Lair keystore : $(lair-keystore --version)"
+              fi
+
+              if ! type "holochain" > /dev/null; then
+                echo "Holochain     : not installed"
+              else
+                echo "Holochain     : $(holochain --version)"
+
+                printf "\nHolochain build info: "
+                holochain --build-info | ${pkgs.jq}/bin/jq
```

Comment:
Just to make the output easier to read, `jq` will format the JSON output and colour it.